src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseResource.java

```diff
@@ -303,9 +303,14 @@ public ResponseEntity<ProgrammingExercise> updateProgrammingExercise(@RequestBod
                 updatedProgrammingExercise.getBuildConfig().isTestwiseCoverageEnabled())) {
             throw new BadRequestAlertException("Testwise coverage enabled flag must not be changed", ENTITY_NAME, "testwiseCoverageCannotChange");
         }
-        if (!Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOnlineEditor()) && !Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOfflineIde())) {
-            return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName,
-                    "You need to allow at least one participation mode, the online editor or the offline IDE", "noParticipationModeAllowed")).body(null);
+        if (!Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOnlineEditor()) && !Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOfflineIde())
```

Comment:
a bit picky, but what happens, when theia is inactive, but for some reason onlineIde is true and the other two values are false? Even if this configuration is not possible in the client, we should validate this in the server

Reply:
Not picky at all - we should validate and sanitize as much as possible imo. Implemented proper checks